elrond-wasm/src/types/token_identifier.rs

```diff
@@ -0,0 +1,194 @@
+use super::BoxedBytes;
+use crate::abi::TypeAbi;
+use alloc::string::String;
+use elrond_codec::*;
+
+/// Specialized type for handling token identifiers.
+/// It wraps a BoxedBytes with the full ASCII name of the token.
+/// EGLD is stored as an empty name.
+///
+/// Not yet implemented, but we might add additional restrictions when deserializing as argument.
+#[derive(Clone, PartialEq, Debug)]
+pub struct TokenIdentifier(BoxedBytes);
+
+impl TokenIdentifier {
+	/// This special representation is interpreted as the EGLD token.
+	pub const EGLD_REPRESENTATION: &'static [u8] = b"EGLD";
+
+	/// New instance of the special EGLD token representation.
+	pub fn egld() -> Self {
+		TokenIdentifier(BoxedBytes::empty())
+	}
+
+	#[inline]
+	pub fn is_egld(&self) -> bool {
+		self.0.is_empty()
+	}
+
+	#[inline]
+	pub fn is_esdt(&self) -> bool {
+		!self.is_egld()
+	}
+
+	#[inline]
+	pub fn into_boxed_bytes(self) -> BoxedBytes {
+		self.0
+	}
```

Comment:
Rename to "as_boxed_bytes" to keep consistency with the other convert methods.

Reply:
`into_boxed_bytes` is actually the correct name here, since it consumes the input (`self` not `&self`)

The convention is outlined here: https://doc.rust-lang.org/1.0.0/style/style/naming/conversions.html